storage_service/locations/models/package.py

```diff
@@ -752,6 +754,104 @@ def backlog_transfer(self, origin_location, origin_path):
         self.status = Package.UPLOADED
         self.save()
 
+    def index_file_data_from_aip_mets(self):
+        """
+        Attempts to read an Archivematica AIP METS file inside this
+        package, then uses the retrieved metadata to generate one entry in the
+        File table in the database for each file inside the package.
+
+        :raises StorageException: if the transfer METS cannot be found,
+            or if required elements are missing.
+        """
+        prefix = self.full_path
+
+        import metsrw
+
+        aip_dir_name = os.path.basename(os.path.splitext(self.full_path)[0])
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Providing a temp dir is unecessary, since `extract_file` creates a temp dir if none is provided, and returns it.  See https://github.com/artefactual/archivematica-storage-service/blob/stable/0.7.x/storage_service/locations/models/package.py#L451-L464